oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/incrementalstore/MergeIncrementalFlatFileStore.java

```diff
@@ -0,0 +1,155 @@
+                        }
+                        baseLine = baseFFSBufferedReader.readLine();
+                        incLine = incrementalFFSBufferedReader.readLine();
+                    }
+                } else {
+                    if (incLine == null) {
+                        writer.write(baseLine);
+                        writer.write("\n");
+                        writeRestOfFile(writer, baseFFSBufferedReader, false);
+                        baseLine = baseFFSBufferedReader.readLine();
+                    } else {
+                        String operand = getOperand(incLine);
+                        if (IncrementalStoreOperand.MODIFY.toString().equals(operand) || IncrementalStoreOperand.DELETE.toString().equals(operand)) {
+                            log.error("incremental ffs should not have modify and delete operands: {}", incLine);
+                            throw new RuntimeException("incremental ffs should not have modify and delete operands" + incLine);
+                        }
+                        writer.write(removeOperand(incLine));
+                        writer.write("\n");
+                        writeRestOfFile(writer, incrementalFFSBufferedReader, true);
+                        incLine = incrementalFFSBufferedReader.readLine();
+                    }
+                }
+            }
+        }
+    }
+
+    private void writeRestOfFile(BufferedWriter writer, BufferedReader bufferedReader, boolean isIncrementalFile) {
+        if (isIncrementalFile) {
+            bufferedReader.lines().forEach((n) -> {
+                String operand = getOperand(n);
+                if (IncrementalStoreOperand.MODIFY.toString().equals(operand) || IncrementalStoreOperand.DELETE.toString().equals(operand)) {
+                    log.error("incremental ffs should not have modify and delete operands: {}", n);
+                    throw new RuntimeException("incremental ffs should not have modify and delete operands" + n);
+                }
+                try {
+                    writer.write(removeOperand(n));
+                    writer.write("\n");
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+        } else {
+            bufferedReader.lines().forEach(n -> {
+                try {
+                    writer.write(n);
+                    writer.write("\n");
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+        }
+    }
+
+    private static String getOperand(String incLine) {
+        String[] parts = NodeStateEntryWriter.getParts(incLine);
+        String operand = parts[2];
+        return operand;
+    }
+
+    private String removeOperand(String line) {
+        String[] parts = NodeStateEntryWriter.getParts(line);
+        return new StringBuilder().append(parts[0]).append("|").append(parts[1]).toString();
```

Comment:
I would avoid to use `StringBuilder` here
```suggestion
        return parts[0] + "|" + parts[1];
```